### PR TITLE
Move Octopus.Shared into Octopus.Tentacle

### PR DESCRIPTION
**Pull request by tothegills**
# Background
This PR moves code from the `Octopus.Shared` project into the `Octopus.Tentacle` project. Also consolidates `Octopus.Shared.Tests` into `Octopus.Tentacle.Tests`. There are some tests that must run on a VM, these have been moved into a new project `Octopus.TentacleTests.Integration`.

The `Octopus.Shared` project remains, containing the script contracts for backwards compatibility.

This PR should not change any functionality.

[Server build](https://github.com/OctopusDeploy/OctopusDeploy/pull/13556) using this Tentacle.

### CODE REVIEW

**Comment by tothegills** [comment]
Fixes https://github.com/OctopusDeploy/ResearchAndDevelopment/issues/486

File: source/Octopus.Tentacle.Tests.Integration/Class1.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    public class Class1
```
Comment (N-lson) [inline source/Octopus.Tentacle.Tests.Integration/Class1.cs:5]: Do we need this?

File: source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs [hunk 2]
```diff
@@ -9,15 +9,15 @@ namespace Octopus.Tentacle.Configuration.Proxy
...
     }
     
     public class ProxyPasswordMaskValuesProvider : IProxyPasswordMaskValuesProvider
     {
         static readonly Regex UrlEncodedCharactersRegex = new Regex(@"%[A-F0-9]{2}", RegexOptions.Compiled);
-        public IEnumerable<string> GetProxyPasswordMaskValues(string proxyPassword)
+        public IEnumerable<string> GetProxyPasswordMaskValues(string? proxyPassword)
         {
-            if (string.IsNullOrEmpty(proxyPassword)) 
+            if (proxyPassword == null || string.IsNullOrEmpty(proxyPassword))
```
Comment (N-lson) [inline source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs:20]: Looks like the existing condition should already check for nullability?

File: source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
 namespace Octopus.Tentacle.Commands
 {
     public class CreateInstanceCommand : AbstractCommand
     {
         readonly IApplicationInstanceManager instanceManager;
-        string instanceName;
-        string config;
-        string home;
+        string instanceName = null!;
```
Comment (N-lson) [inline source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs:10]: Do we want to revisit these `null!` at some point to clean them up? Think they're fine to keep in as part of this PR.

File: source/Octopus.Tentacle/Octopus.Tentacle.csproj [hunk 4]
```diff
@@ -13,6 +13,7 @@
 		<TargetFrameworks>net452;netcoreapp3.1</TargetFrameworks>
 		<TreatWarningsAsErrors>true</TreatWarningsAsErrors>
 		<LangVersion>9</LangVersion>
+		<Nullable>enable</Nullable>
```
Comment (N-lson) [inline source/Octopus.Tentacle/Octopus.Tentacle.csproj:16]: 👍

**Review by N-lson — APPROVED** [review]
Looks like a good move 👍 things that aren't shared make more sense belonging in `Octopus.Tentacle` itself.

File: source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs (hunk 3 above)
Comment (tothegills) [reply source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs:10]: Yes, I might change these back for the test project so they are easier to spot if we flip nullable on.

File: source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs (hunk 2 above)
Comment (tothegills) [reply source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs:20]: It doesn't and results in a compiler error.

Comment (N-lson) [reply source/Octopus.Tentacle/Configuration/Proxy/ProxyPasswordMaskValuesProvider.cs:20]: damn

File: source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs (hunk 3 above)
Comment (tothegills) [reply source/Octopus.Tentacle/Commands/CreateInstanceCommand.cs:10]: There is some "prior art" in the tests that uses nullable and `null!` initializers. Will leave them as-is for now, tech-debt to be paid off in the future.

**Pull request merged**